From ac1513cd1a1b7341ff2e90651ce5ceeadc5e707e Mon Sep 17 00:00:00 2001
From: David Dollar <ddollar@gmail.com>
Date: Sat, 14 Mar 2009 11:43:51 -0400
Subject: [PATCH 2/2] Additional tests for the gem subsystem

  * test_gem_ignores_development_dependencies
  * test_gem_guards_against_duplicate_unpacks
  * test_gem_does_not_unpack_framework_gems
---
 railties/test/gem_dependency_test.rb               |   20 ++++++++
 .../vendor/gems/dummy-gem-h-1.0.0/.specification   |   29 ++++++++++++
 .../gems/dummy-gem-h-1.0.0/lib/dummy-gem-h.rb      |    1 +
 .../vendor/gems/dummy-gem-i-1.0.0/.specification   |   49 ++++++++++++++++++++
 .../gems/dummy-gem-i-1.0.0/lib/dummy-gem-i.rb      |    1 +
 5 files changed, 100 insertions(+), 0 deletions(-)
 create mode 100644 railties/test/vendor/gems/dummy-gem-h-1.0.0/.specification
 create mode 100644 railties/test/vendor/gems/dummy-gem-h-1.0.0/lib/dummy-gem-h.rb
 create mode 100644 railties/test/vendor/gems/dummy-gem-i-1.0.0/.specification
 create mode 100644 railties/test/vendor/gems/dummy-gem-i-1.0.0/lib/dummy-gem-i.rb

diff --git a/railties/test/gem_dependency_test.rb b/railties/test/gem_dependency_test.rb
index 189ad02..a44b6a8 100644
--- a/railties/test/gem_dependency_test.rb
+++ b/railties/test/gem_dependency_test.rb
@@ -144,4 +144,24 @@ class GemDependencyTest < Test::Unit::TestCase
     end
   end
 
+  def test_gem_ignores_development_dependencies
+    dummy_gem = Rails::GemDependency.new "dummy-gem-i"
+    dummy_gem.add_load_paths
+    dummy_gem.load
+    assert_equal 1, dummy_gem.dependencies.size
+  end
+
+  def test_gem_guards_against_duplicate_unpacks
+    dummy_gem = Rails::GemDependency.new "dummy-gem-a"
+    dummy_gem.expects(:frozen?).returns(true)
+    dummy_gem.expects(:unpack_base).never
+    dummy_gem.unpack
+  end
+
+  def test_gem_does_not_unpack_framework_gems
+    dummy_gem = Rails::GemDependency.new "dummy-gem-a"
+    dummy_gem.expect(:framework_gem?).returns(true)
+    dummy_gem.expects(:unpack_base).never
+    dummy_gem.unpack
+  end
 end
diff --git a/railties/test/vendor/gems/dummy-gem-h-1.0.0/.specification b/railties/test/vendor/gems/dummy-gem-h-1.0.0/.specification
new file mode 100644
index 0000000..b3f7930
--- /dev/null
+++ b/railties/test/vendor/gems/dummy-gem-h-1.0.0/.specification
@@ -0,0 +1,29 @@
+--- !ruby/object:Gem::Specification
+name: dummy-gem-h
+version: !ruby/object:Gem::Version
+  version: 1.3.0
+platform: ruby
+authors:
+- "Nobody"
+date: 2008-10-03 00:00:00 -04:00
+dependencies:
+files:
+- lib
+- lib/dummy-gem-h.rb
+require_paths:
+- lib
+required_ruby_version: !ruby/object:Gem::Requirement
+  requirements:
+  - - ">="
+    - !ruby/object:Gem::Version
+      version: "0"
+  version:
+required_rubygems_version: !ruby/object:Gem::Requirement
+  requirements:
+  - - ">="
+    - !ruby/object:Gem::Version
+      version: "0"
+  version:
+requirements: []
+specification_version: 2
+summary: Dummy Gem H
diff --git a/railties/test/vendor/gems/dummy-gem-h-1.0.0/lib/dummy-gem-h.rb b/railties/test/vendor/gems/dummy-gem-h-1.0.0/lib/dummy-gem-h.rb
new file mode 100644
index 0000000..0f91234
--- /dev/null
+++ b/railties/test/vendor/gems/dummy-gem-h-1.0.0/lib/dummy-gem-h.rb
@@ -0,0 +1 @@
+DUMMY_GEM_H_VERSION="1.0.0"
diff --git a/railties/test/vendor/gems/dummy-gem-i-1.0.0/.specification b/railties/test/vendor/gems/dummy-gem-i-1.0.0/.specification
new file mode 100644
index 0000000..f8af41d
--- /dev/null
+++ b/railties/test/vendor/gems/dummy-gem-i-1.0.0/.specification
@@ -0,0 +1,49 @@
+--- !ruby/object:Gem::Specification
+name: dummy-gem-i
+version: !ruby/object:Gem::Version
+  version: 1.3.0
+platform: ruby
+authors:
+- "Nobody"
+date: 2008-10-03 00:00:00 -04:00
+dependencies:
+- !ruby/object:Gem::Dependency
+  name: dummy-gem-f
+  type: :runtime
+  version_requirement:
+  version_requirements: !ruby/object:Gem::Requirement
+    requirements:
+    - - ">="
+      - !ruby/object:Gem::Version
+        version: 1.0.0
+    version:
+- !ruby/object:Gem::Dependency
+  name: dummy-gem-h
+  type: :development
+  version_requirement:
+  version_requirements: !ruby/object:Gem::Requirement
+    requirements:
+    - - ">="
+      - !ruby/object:Gem::Version
+        version: 1.0.0
+    version:
+files:
+- lib
+- lib/dummy-gem-i.rb
+require_paths:
+- lib
+required_ruby_version: !ruby/object:Gem::Requirement
+  requirements:
+  - - ">="
+    - !ruby/object:Gem::Version
+      version: "0"
+  version:
+required_rubygems_version: !ruby/object:Gem::Requirement
+  requirements:
+  - - ">="
+    - !ruby/object:Gem::Version
+      version: "0"
+  version:
+requirements: []
+specification_version: 2
+summary: Dummy Gem I
diff --git a/railties/test/vendor/gems/dummy-gem-i-1.0.0/lib/dummy-gem-i.rb b/railties/test/vendor/gems/dummy-gem-i-1.0.0/lib/dummy-gem-i.rb
new file mode 100644
index 0000000..2f9a376
--- /dev/null
+++ b/railties/test/vendor/gems/dummy-gem-i-1.0.0/lib/dummy-gem-i.rb
@@ -0,0 +1 @@
+DUMMY_GEM_I_VERSION="1.0.0"
-- 
1.6.1.2

